src/services/navigation/useNavLinks/useNavLinks.js

```diff
@@ -382,12 +382,17 @@ export function useNavLinks() {
     },
     componentsTab: {
       path: (
```

Comment:
In the navlink to go to components tab on repo page, add an optional query param to select the initial branch to load, this is used by PR page components table where it links to the repo page components table while keeping the associated branch of them PR.